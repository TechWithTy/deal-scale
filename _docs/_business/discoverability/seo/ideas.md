# DealScale SEO Vision & Opportunity Map

## Purpose
Capture strategic opportunities to elevate DealScale's organic visibility across www, blog, and app surfaces while aligning with technical capabilities.

## Guiding Principles
- **Entity-First Architecture:** All content should reinforce the canonical Organization graph node at `https://www.dealscale.io/#organization`.
- **Multi-Domain Cohesion:** Ensure messaging and metadata stay synchronized across www, blog, and app domains.
- **Incremental Delivery:** Ship schema coverage, crawl control, and content improvements in progressive slices tied to measurable KPIs.
- **Data-Informed Iteration:** Pair every SEO enhancement with analytics hooks and validation paths.

## Opportunity Themes

### 1. Crawl Management & Discoverability
- Expand `robots.txt` directives for priority crawlers (Googlebot, GPTBot, PerplexityBot, ClaudeBot).
- Establish XML sitemap segmentation for events, partners, careers, and cornerstone resources.
- Automate sitemap submission to Google Search Console during deployments.

### 2. Structured Data Foundation
- Consolidate JSON-LD builders inside a shared `seo-core` utility layer.
- Standardize schema usage for Organization, WebSite, Event, ItemList, Product/Service, CreativeWork, FAQPage, and JobPosting.
- Introduce schema regression testing to guard against accidental field removals.

### 3. Route-Level Enhancements
- Convert `/events` to ISR-backed pages with canonical metadata and event-specific JSON-LD.
- Add `/partners` gallery with ItemList schema referencing partner organizations.
- Launch `/careers` with Zoho Recruit integration, JobPosting schema, and long-tail job content.
- Enrich `/case-studies/[slug]` with CreativeWork + Review schema when testimonials exist.

### 4. Content & UX Synergy
- Publish evergreen content pillars (Features, Pricing, Case Studies) with FAQ sections for rich snippets.
- Integrate internal linking modules highlighting events, partners, and jobs to distribute link equity.
- Refresh on-page copy to emphasize target keywords and intent clusters discovered via competitor analysis.

### 5. Measurement & Feedback Loops
- Track SEO KPIs (organic sessions, CTR, schema validation success) through PostHog dashboards.
- Run quarterly Lighthouse audits targeting 95+ SEO and 90+ performance scores.
- Document QA flows for Google Rich Results, schema validation, and bot fetch snapshots.

## Experiment Backlog
1. **Dynamic FAQ Blocks:** Inject Zod-validated FAQ data into `/features` and `/pricing` pages.
2. **VideoObject Schema:** Tag product demos and webinars to surface in video SERPs.
3. **Site Search Action:** Add `potentialAction` to WebSite schema to improve sitelinks search box eligibility.
4. **Partner Co-Marketing Pages:** Spin up co-branded landing pages with reciprocal backlinks.
5. **Localized Landing Pages:** Evaluate region-specific variations once global expansion matures.
