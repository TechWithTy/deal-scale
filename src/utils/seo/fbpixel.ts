declare global {
	interface Window {
		fbq?: (...args: unknown[]) => void;
	}
}

export const FB_PIXEL_ID = process.env.NEXT_PUBLIC_FACEBOOK_PIXEL_ID;

export const pageView = (properties?: Record<string, unknown>) => {
	if (window.fbq) window.fbq("track", "PageView", properties);
};

export const event = (name: string, options: Record<string, unknown> = {}) => {
	if (window.fbq) window.fbq("track", name, options);
};
