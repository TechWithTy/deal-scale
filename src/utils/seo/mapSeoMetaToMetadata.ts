import type { Metadata } from "next";
// * Maps custom SeoMeta to Next.js Metadata type, including canonical tag
import type { SeoMeta } from "./seo";

export function mapSeoMetaToMetadata(seo: SeoMeta): Metadata {
	// Ensure we have a full URL for the image
	const imageUrl = seo.image?.startsWith("http")
		? seo.image
		: `${process.env.NEXT_PUBLIC_SITE_URL || "https://dealscale.io"}${seo.image.startsWith("/") ? "" : "/"}${seo.image}`;

	return {
		title: seo.title,
		description: seo.description,
		keywords: seo.keywords,
		openGraph: {
			title: seo.title,
			description: seo.description,
			url: seo.canonical,
			siteName: seo.siteName,
			images: seo.image
				? [
						{
							url: imageUrl,
							width: 1200,
							height: 630,
							alt: seo.title,
						},
					]
				: undefined,
			type: seo.type || "website",
		},
		twitter: seo.image
			? {
					card: "summary_large_image",
					images: [imageUrl],
					title: seo.title,
					description: seo.description,
				}
			: undefined,
		alternates: {
			canonical: seo.canonical,
		},
	};
}
