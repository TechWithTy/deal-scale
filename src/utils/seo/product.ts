import { mockProducts } from "@/data/products";
import type { ProductType } from "@/types/products";
import { getTestBaseUrl } from "@/utils/env";
import type { SeoMeta } from "@/utils/seo/seo";

/**
 * Fetch a product by slug (replace with real DB or API call in production)
 */
function getProductBySlug(slug: string): ProductType | undefined {
	return mockProducts.find((p) => p.sku.toLowerCase() === slug.toLowerCase());
}

/**
 * Generate dynamic SEO metadata for a product page
 */
export async function getSeoMetadataForProduct(slug: string): Promise<SeoMeta> {
	const product = getProductBySlug(slug);
	const baseUrl = getTestBaseUrl();
	const pageUrl = `${baseUrl}/products/${slug}`;

	if (!product) {
		return {
			title: "Product Not Found",
			description: "The requested product could not be found.",
			canonical: pageUrl,
			keywords: [],
			image: "",
			type: "article",
			priority: 0.6, // * fallback
			changeFrequency: "monthly", // * fallback
		};
	}

	// Compose keywords from product name, type, and default keywords
	const keywords = [
		product.name,
		...(product.types?.map((t) => t.name) || []),
		"Deal Scale",
		"shop",
		"products",
		"digital",
		"ecommerce",
	];

	return {
		title: `${product.name} | Product | Deal Scale`,
		description: product.description || "Product details and features.",
		canonical: pageUrl,
		keywords,
		image: product.images?.[0] || "",
		type: "article",
		priority: 0.6, // * or customize per product
		changeFrequency: "monthly", // * or customize per product
	};
}
