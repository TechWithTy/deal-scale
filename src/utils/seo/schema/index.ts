export { SchemaInjector } from "./SchemaInjector";
export {
        buildBlogSchema,
        buildCaseStudyCreativeWorkSchema,
        buildOrganizationSchema,
        buildProductSchema,
        buildFAQPageSchema,
        buildServiceSchema,
        buildWebSiteSchema,
} from "./builders";
export {
        buildAbsoluteUrl,
        buildContactPoints,
        buildPostalAddress,
        buildSocialProfiles,
        ORGANIZATION_ID,
        SCHEMA_CONTEXT,
        WEBSITE_ID,
} from "./helpers";
export type {
        OfferInput,
        OfferSchema,
        OrganizationSchema,
        BlogSchema,
        BlogPostingSchema,
        CreativeWorkSchema,
        FAQPageSchema,
        ProductSchema,
        ProductSchemaInput,
        ReviewSchema,
        ServiceSchema,
        ServiceSchemaInput,
        WebSiteSchema,
        ContactPointSchema,
        PostalAddressSchema,
        SchemaContext,
} from "./types";
