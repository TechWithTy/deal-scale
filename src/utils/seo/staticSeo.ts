import { DEFAULT_SEO, STATIC_SEO_META } from "@/data/constants/seo";
import type { SeoMeta } from "./seo";

export const defaultSeo: SeoMeta = {
	title: DEFAULT_SEO.title,
	description: DEFAULT_SEO.description,
	canonical: DEFAULT_SEO.canonical,
	keywords: DEFAULT_SEO.keywords,
	image: DEFAULT_SEO.image,
	siteName: DEFAULT_SEO.siteName,
	type: DEFAULT_SEO.type,
};

export const staticSeoMeta: Record<string, SeoMeta> = {
	"/": {
		title: STATIC_SEO_META["/"].title,
		description: STATIC_SEO_META["/"].description,
		canonical: STATIC_SEO_META["/"].canonical,
		image: STATIC_SEO_META["/"].image,
		keywords: DEFAULT_SEO.keywords,
		priority: 1,
		changeFrequency: "daily",
		siteName: DEFAULT_SEO.siteName,
		type: DEFAULT_SEO.type,
	},
	"/products": STATIC_SEO_META["/products"] ?? {
		priority: 0.8,
		changeFrequency: "weekly",
		title: "Products | Deal Scale",
		description:
			"Browse our curated collection of digital and physical products from Deal Scale.",
		canonical: "https://dealscale.io/products",
		image: DEFAULT_SEO.image,
		keywords: [...DEFAULT_SEO.keywords, "products", "shop", "catalog"],
		siteName: DEFAULT_SEO.siteName,
		type: DEFAULT_SEO.type,
	},
	"/privacy": {
		priority: 0.5,
		changeFrequency: "monthly",
		title: STATIC_SEO_META["/privacy"].title,
		description: STATIC_SEO_META["/privacy"].description,
		canonical: STATIC_SEO_META["/privacy"].canonical,
		image: STATIC_SEO_META["/privacy"].image,
		keywords: DEFAULT_SEO.keywords,
	},
	"/blogs": {
		priority: 0.8,
		changeFrequency: "weekly",
		title: STATIC_SEO_META["/blogs"].title,
		description: STATIC_SEO_META["/blogs"].description,
		canonical: STATIC_SEO_META["/blogs"].canonical,
		image: STATIC_SEO_META["/blogs"].image,
		keywords: DEFAULT_SEO.keywords,
	},
	"/case-studies": {
		priority: 0.8,
		changeFrequency: "weekly",
		title: STATIC_SEO_META["/case-studies"].title,
		description: STATIC_SEO_META["/case-studies"].description,
		canonical: STATIC_SEO_META["/case-studies"].canonical,
		image: STATIC_SEO_META["/case-studies"].image,
		keywords: DEFAULT_SEO.keywords,
	},
	"/contact": {
		priority: 0.7,
		changeFrequency: "monthly",
		title: STATIC_SEO_META["/contact"].title,
		description: STATIC_SEO_META["/contact"].description,
		canonical: STATIC_SEO_META["/contact"].canonical,
		image: STATIC_SEO_META["/contact"].image,
		keywords: DEFAULT_SEO.keywords,
	},
	"/contact-pilot": {
		priority: 0.7,
		changeFrequency: "monthly",
		title: STATIC_SEO_META["/contact-pilot"].title,
		description: STATIC_SEO_META["/contact-pilot"].description,
		canonical: STATIC_SEO_META["/contact-pilot"].canonical,
		image: STATIC_SEO_META["/contact-pilot"].image,
		keywords: DEFAULT_SEO.keywords,
	},
	"/cookies": {
		priority: 0.3,
		changeFrequency: "yearly",
		title: STATIC_SEO_META["/cookies"].title,
		description: STATIC_SEO_META["/cookies"].description,
		canonical: STATIC_SEO_META["/cookies"].canonical,
		image: STATIC_SEO_META["/cookies"].image,
		keywords: DEFAULT_SEO.keywords,
	},
        "/events": {
                title: STATIC_SEO_META["/events"].title,
                description: STATIC_SEO_META["/events"].description,
                canonical: STATIC_SEO_META["/events"].canonical,
                image: STATIC_SEO_META["/events"].image,
                keywords: DEFAULT_SEO.keywords,
                priority: 0.8,
                changeFrequency: "weekly",
        },
        "/faqs": {
                title: STATIC_SEO_META["/faqs"].title,
                description: STATIC_SEO_META["/faqs"].description,
                canonical: STATIC_SEO_META["/faqs"].canonical,
                image: STATIC_SEO_META["/faqs"].image,
                keywords: DEFAULT_SEO.keywords,
        },
        "/partners": {
                title: STATIC_SEO_META["/partners"].title,
                description: STATIC_SEO_META["/partners"].description,
                canonical: STATIC_SEO_META["/partners"].canonical,
                image: STATIC_SEO_META["/partners"].image,
                keywords: DEFAULT_SEO.keywords,
                priority: 0.7,
                changeFrequency: "monthly",
        },
        "/careers": {
                title: STATIC_SEO_META["/careers"].title,
                description: STATIC_SEO_META["/careers"].description,
                canonical: STATIC_SEO_META["/careers"].canonical,
                image: STATIC_SEO_META["/careers"].image,
                keywords: DEFAULT_SEO.keywords,
                priority: 0.65,
                changeFrequency: "weekly",
        },
        "/pricing": {
                title: STATIC_SEO_META["/pricing"].title,
                description: STATIC_SEO_META["/pricing"].description,
                canonical: STATIC_SEO_META["/pricing"].canonical,
                image: STATIC_SEO_META["/pricing"].image,
		keywords: DEFAULT_SEO.keywords,
	},
	"/portfolio": {
		title: STATIC_SEO_META["/portfolio"].title,
		description: STATIC_SEO_META["/portfolio"].description,
		canonical: STATIC_SEO_META["/portfolio"].canonical,
		image: STATIC_SEO_META["/portfolio"].image,
		keywords: DEFAULT_SEO.keywords,
	},
	"/features": {
		title: STATIC_SEO_META["/features"].title,
		description: STATIC_SEO_META["/features"].description,
		canonical: STATIC_SEO_META["/features"].canonical,
		image: STATIC_SEO_META["/features"].image,
		keywords: DEFAULT_SEO.keywords,
	},
	"/tos": {
		title: STATIC_SEO_META["/tos"].title,
		description: STATIC_SEO_META["/tos"].description,
		canonical: STATIC_SEO_META["/tos"].canonical,
		image: STATIC_SEO_META["/tos"].image,
		keywords: DEFAULT_SEO.keywords,
	},
	"/GDPR": {
		title: STATIC_SEO_META["/GDPR"].title,
		description: STATIC_SEO_META["/GDPR"].description,
		canonical: STATIC_SEO_META["/GDPR"].canonical,
		image: STATIC_SEO_META["/GDPR"].image,
		keywords: DEFAULT_SEO.keywords,
	},
	"/hippa": {
		title: STATIC_SEO_META["/hippa"].title,
		description: STATIC_SEO_META["/hippa"].description,
		canonical: STATIC_SEO_META["/hippa"].canonical,
		image: STATIC_SEO_META["/hippa"].image,
		keywords: DEFAULT_SEO.keywords,
	},
	"/PII": {
		title: STATIC_SEO_META["/PII"].title,
		description: STATIC_SEO_META["/PII"].description,
		canonical: STATIC_SEO_META["/PII"].canonical,
		image: STATIC_SEO_META["/PII"].image,
		keywords: DEFAULT_SEO.keywords,
	},
	"/tcpCompliance": {
		title: STATIC_SEO_META["/tcpCompliance"].title,
		description: STATIC_SEO_META["/tcpCompliance"].description,
		canonical: STATIC_SEO_META["/tcpCompliance"].canonical,
		image: STATIC_SEO_META["/tcpCompliance"].image,
		keywords: DEFAULT_SEO.keywords,
	},
	"/legal": {
		title: STATIC_SEO_META["/legal"].title,
		description: STATIC_SEO_META["/legal"].description,
		canonical: STATIC_SEO_META["/legal"].canonical,
		image: STATIC_SEO_META["/legal"].image,
		keywords: DEFAULT_SEO.keywords,
	},
	"/linktree": {
		title: STATIC_SEO_META["/linktree"].title,
		description: STATIC_SEO_META["/linktree"].description,
		canonical: STATIC_SEO_META["/linktree"].canonical,
		image: STATIC_SEO_META["/linktree"].image,
		keywords: STATIC_SEO_META["/linktree"].keywords || DEFAULT_SEO.keywords,
		priority: 0.7,
		changeFrequency: "daily", // Links update frequently from Notion
	},
};

export function getStaticSeo(path: string): SeoMeta {
	return staticSeoMeta[path] || defaultSeo;
}
