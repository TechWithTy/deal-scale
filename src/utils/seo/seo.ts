import type { MediumArticle } from "@/data/medium/post";
import type { BeehiivPost } from "@/types/behiiv";
import type { CaseStudy } from "@/types/case-study";
import type { ServiceItemData } from "@/types/service/services";
import * as React from "react";
import { defaultSeo } from "./staticSeo";

import { getPostByGuid } from "@/lib/medium/get";
/**
 * Returns full Next.js Metadata for a blog post, including canonical/original tags
 * Used by both generateMetadata and test suites for DRY SEO logic
 */
// src/utils/seo/getSeoMetadataForPost.ts
import type { Metadata } from "next";

export interface SeoMeta {
	/**
	 * Sitemap priority for SEO (0.0 - 1.0)
	 */
	priority?: number;
	/**
	 * Sitemap changeFrequency for SEO
	 */
	changeFrequency?:
		| "always"
		| "hourly"
		| "daily"
		| "weekly"
		| "monthly"
		| "yearly"
		| "never";
	title: string;
	description: string;
	canonical?: string;
	keywords: string[];
	image: string;
	type?:
		| "website"
		| "article"
		| "book"
		| "profile"
		| "music.song"
		| "music.album"
		| "music.playlist"
		| "music.radio_station"
		| "video.movie"
		| "video.episode"
		| "video.tv_show"
		| "video.other";
	datePublished?: string;
	dateModified?: string;
	siteName?: string;
	[key: string]:
		| string
		| string[]
		| number
		| SeoMeta["changeFrequency"]
		| undefined;
}

// * Renders Open Graph meta tags for Next.js/React head
// ! Only use in server components or <head> blocks

export function renderOpenGraphMeta(openGraph: {
	title?: string;
	description?: string;
	url?: string;
	type?:
		| "website"
		| "article"
		| "book"
		| "profile"
		| "music.song"
		| "music.album"
		| "music.playlist"
		| "music.radio_station"
		| "video.movie"
		| "video.episode"
		| "video.tv_show"
		| "video.other";
	publishedTime?: string;
	authors?: string[];
	images?: { url: string; width?: number; height?: number; alt?: string }[];
	keywords?: string[];
	siteName?: string;
	locale?: string;
	metadataBase?: string;
	twitter?: {
		card?: string;
		title?: string;
		description?: string;
		creator?: string;
		images?: string[];
	};
}): React.ReactNode[] {
	if (!openGraph) return [];
	const tags: React.ReactNode[] = [];
	if (openGraph.title)
		tags.push(
			React.createElement("meta", {
				key: "og:title",
				property: "og:title",
				content: openGraph.title,
			}),
		);
	if (openGraph.description)
		tags.push(
			React.createElement("meta", {
				key: "og:description",
				property: "og:description",
				content: openGraph.description,
			}),
		);
	if (openGraph.url)
		tags.push(
			React.createElement("meta", {
				key: "og:url",
				property: "og:url",
				content: openGraph.url,
			}),
		);
	if (openGraph.type)
		tags.push(
			React.createElement("meta", {
				key: "og:type",
				property: "og:type",
				content: openGraph.type,
			}),
		);
	if (openGraph.publishedTime)
		tags.push(
			React.createElement("meta", {
				key: "og:published_time",
				property: "og:published_time",
				content: openGraph.publishedTime,
			}),
		);
	if (openGraph.authors?.length)
		openGraph.authors.forEach((author, i) =>
			tags.push(
				React.createElement("meta", {
					key: `og:article:author:${i}`,
					property: "article:author",
					content: author,
				}),
			),
		);
	if (openGraph.images?.length)
		openGraph.images.forEach((img, i) => {
			tags.push(
				React.createElement("meta", {
					key: `og:image:${i}`,
					property: "og:image",
					content: img.url,
				}),
			);
			if (img.width)
				tags.push(
					React.createElement("meta", {
						key: `og:image:width:${i}`,
						property: "og:image:width",
						content: String(img.width),
					}),
				);
			if (img.height)
				tags.push(
					React.createElement("meta", {
						key: `og:image:height:${i}`,
						property: "og:image:height",
						content: String(img.height),
					}),
				);
			if (img.alt)
				tags.push(
					React.createElement("meta", {
						key: `og:image:alt:${i}`,
						property: "og:image:alt",
						content: img.alt,
					}),
				);
		});
	// Keywords
	if (openGraph.keywords?.length) {
		tags.push(
			React.createElement("meta", {
				key: "keywords",
				name: "keywords",
				content: openGraph.keywords.join(", "),
			}),
		);
	}
	// OG site name
	if (openGraph.siteName)
		tags.push(
			React.createElement("meta", {
				key: "og:site_name",
				property: "og:site_name",
				content: openGraph.siteName,
			}),
		);
	// OG locale
	if (openGraph.locale)
		tags.push(
			React.createElement("meta", {
				key: "og:locale",
				property: "og:locale",
				content: openGraph.locale,
			}),
		);
	// OG metadataBase as canonical url
	if (openGraph.metadataBase)
		tags.push(
			React.createElement("link", {
				key: "canonical",
				rel: "canonical",
				href: openGraph.metadataBase,
			}),
		);
	// Twitter meta tags
	if (openGraph.twitter) {
		if (openGraph.twitter.card)
			tags.push(
				React.createElement("meta", {
					key: "twitter:card",
					name: "twitter:card",
					content: openGraph.twitter.card,
				}),
			);
		if (openGraph.twitter.title)
			tags.push(
				React.createElement("meta", {
					key: "twitter:title",
					name: "twitter:title",
					content: openGraph.twitter.title,
				}),
			);
		if (openGraph.twitter.description)
			tags.push(
				React.createElement("meta", {
					key: "twitter:description",
					name: "twitter:description",
					content: openGraph.twitter.description,
				}),
			);
		if (openGraph.twitter.creator)
			tags.push(
				React.createElement("meta", {
					key: "twitter:creator",
					name: "twitter:creator",
					content: openGraph.twitter.creator,
				}),
			);
		if (openGraph.twitter.images?.length)
			openGraph.twitter.images.forEach((img, i) =>
				tags.push(
					React.createElement("meta", {
						key: `twitter:image:${i}`,
						name: "twitter:image",
						content: img,
					}),
				),
			);
	}
	return tags;
}
export function getBlogSeo(post: BeehiivPost): SeoMeta {
	// Description: prefer meta_default_description, preview_text, or content
	let description =
		post.meta_default_description ||
		post.preview_text ||
		defaultSeo.description;
	if (
		!description &&
		post.content &&
		typeof post.content === "object" &&
		"free" in post.content &&
		typeof post.content.free.web === "string"
	) {
		description = post.content.free.web.slice(0, 160);
	}

	// Keywords: use content_tags or fallback
	const keywords =
		Array.isArray(post.content_tags) && post.content_tags.length > 0
			? post.content_tags
			: defaultSeo.keywords;

	// Image: use thumbnail_url or fallback
	const image = post.thumbnail_url || defaultSeo.image;

	// Dates: use publish_date if present, else undefined
	const datePublished =
		typeof post.publish_date === "number"
			? new Date(post.publish_date).toISOString()
			: undefined;
	const dateModified =
		typeof post.displayed_date === "number"
			? new Date(post.displayed_date).toISOString()
			: datePublished;

	return {
		title: `${post.title} | Blog | Deal Scale`,
		description,
		canonical: post.web_url || undefined,
		keywords,
		image,
		type: "article",
		datePublished,
		dateModified,
	};
}

export function getCaseStudySeo(cs: CaseStudy): SeoMeta {
	return {
		title: `${cs.title} | Case Study | Deal Scale`,
		description: cs.subtitle || defaultSeo.description,
		canonical: `https://dealscale.io/case-studies/${cs.slug}`,
		keywords: defaultSeo.keywords,
		image: cs.featuredImage || defaultSeo.image,
		type: "article",
		datePublished: cs.lastModified.toISOString(),
		dateModified: cs.lastModified.toISOString(),
	};
}
export function getServiceSeo(service: ServiceItemData): SeoMeta {
	return {
		title: `${service.title} | Services | Deal Scale`,
		description: service.description || defaultSeo.description,
		canonical: `https://dealscale.io/features/${service.slugDetails.slug}`,
		keywords: defaultSeo.keywords,
		image: defaultSeo.image,
		type: "article",
		datePublished: service.slugDetails.datePublished
			? service.slugDetails.datePublished.toISOString()
			: new Date().toISOString(),
		dateModified: service.slugDetails.lastModified
			? service.slugDetails.lastModified.toISOString()
			: new Date().toISOString(),
	};
}
