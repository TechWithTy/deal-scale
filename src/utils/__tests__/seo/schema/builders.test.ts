import {
        buildOrganizationSchema,
        buildProductSchema,
        buildServiceSchema,
        buildWebSiteSchema,
} from "@/utils/seo/schema";
import { defaultSeo } from "@/utils/seo/staticSeo";

describe("schema builders", () => {
        it("builds the organization schema", () => {
                expect(buildOrganizationSchema()).toMatchSnapshot();
        });

        it("builds the website schema", () => {
                expect(buildWebSiteSchema()).toMatchSnapshot();
        });

        it("builds a service schema", () => {
                const serviceSchema = buildServiceSchema({
                        name: "AI Appointment Setting",
                        description:
                                "Done-for-you lead qualification powered by Deal Scale's AI agents.",
                        url: `${defaultSeo.canonical}/services/ai-appointment-setting`,
                        serviceType: "Lead Generation",
                        category: "AI Services",
                });

                expect(serviceSchema).toMatchSnapshot();
        });

        it("builds a product schema", () => {
                const productSchema = buildProductSchema({
                        name: "AI Lead Nurture Toolkit",
                        description:
                                "A comprehensive toolkit for managing AI-assisted lead nurture workflows.",
                        url: `${defaultSeo.canonical}/products/ai-lead-nurture-toolkit`,
                        sku: "AI-LEAD-001",
                        offers: {
                                price: 199,
                                priceCurrency: "USD",
                        },
                });

                expect(productSchema).toMatchSnapshot();
        });
});
