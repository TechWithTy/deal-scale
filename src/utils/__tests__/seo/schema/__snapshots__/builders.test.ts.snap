// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`schema builders builds a product schema 1`] = `
{
  "@context": "https://schema.org",
  "@id": "https://dealscale.io/products/ai-lead-nurture-toolkit#product",
  "@type": "Product",
  "brand": {
    "@type": "Brand",
    "name": "CyberOni",
  },
  "description": "A comprehensive toolkit for managing AI-assisted lead nurture workflows.",
  "image": undefined,
  "manufacturer": {
    "@id": "https://dealscale.io#organization",
  },
  "name": "AI Lead Nurture Toolkit",
  "offers": {
    "@type": "Offer",
    "price": 199,
    "priceCurrency": "USD",
    "url": undefined,
  },
  "sku": "AI-LEAD-001",
  "url": "https://dealscale.io/products/ai-lead-nurture-toolkit",
}
`;

exports[`schema builders builds a service schema 1`] = `
{
  "@context": "https://schema.org",
  "@id": "https://dealscale.io/services/ai-appointment-setting#service",
  "@type": "Service",
  "areaServed": undefined,
  "category": "AI Services",
  "description": "Done-for-you lead qualification powered by Deal Scale's AI agents.",
  "name": "AI Appointment Setting",
  "offers": undefined,
  "provider": {
    "@id": "https://dealscale.io#organization",
  },
  "serviceType": "Lead Generation",
  "url": "https://dealscale.io/services/ai-appointment-setting",
}
`;

exports[`schema builders builds the organization schema 1`] = `
{
  "@context": "https://schema.org",
  "@id": "https://dealscale.io#organization",
  "@type": "Organization",
  "address": {
    "@type": "PostalAddress",
    "addressLocality": "Denver",
    "addressRegion": "CO",
    "streetAddress": "3700 Quebec St",
  },
  "contactPoint": [
    {
      "@type": "ContactPoint",
      "availableLanguage": [
        "en",
      ],
      "contactType": "customer service",
      "telephone": "+1 (720) 258-6576",
    },
    {
      "@type": "ContactPoint",
      "availableLanguage": [
        "en",
      ],
      "contactType": "sales",
      "email": "sam.scaler@dealscale.io",
    },
  ],
  "description": "Deal Scale provides AI-Powered Agents that deliver sales-ready appointments, not just leads. We automate the repetitive, time-consuming work of prospecting and nurturing so real estate investors and wholesalers can focus on what matters most: closing deals. Our mission is to build a consistent, predictable pipeline for every client.",
  "legalName": "CyberOni",
  "logo": "https://dealscale.io/banners/main.png",
  "name": "CyberOni",
  "sameAs": [
    "https://www.linkedin.com/company/deal-scale/",
    "https://www.facebook.com/profile.php?id=61576707389189",
    "https://www.instagram.com/deal_scale/",
    "https://medium.com/@dealscale",
  ],
  "url": "https://dealscale.io",
}
`;

exports[`schema builders builds the website schema 1`] = `
{
  "@context": "https://schema.org",
  "@id": "https://dealscale.io#website",
  "@type": "WebSite",
  "description": "Expert AI-powered solutions for real estate agents, investors and wholesalers. Automate lead generation, qualification, and outreach to close more deals.",
  "name": "Deal Scale | AI-Powered Real Estate Solutions",
  "potentialAction": {
    "@type": "SearchAction",
    "query-input": "required name=search_term_string",
    "target": "https://dealscale.io/search?q={search_term_string}",
  },
  "publisher": {
    "@id": "https://dealscale.io#organization",
  },
  "url": "https://dealscale.io",
}
`;
